core/src/integration-test/java/com/scalar/db/api/TwoPhaseCommitTransactionIntegrationTestBase.java

```diff
@@ -0,0 +1,810 @@
+package com.scalar.db.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import com.scalar.db.api.Scan.Ordering;
+import com.scalar.db.config.DatabaseConfig;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.transaction.CommitException;
+import com.scalar.db.exception.transaction.CrudException;
+import com.scalar.db.exception.transaction.PreparationConflictException;
+import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.db.io.DataType;
+import com.scalar.db.io.IntValue;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.Value;
+import com.scalar.db.service.TransactionFactory;
+import com.scalar.db.util.TestUtils;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class TwoPhaseCommitTransactionIntegrationTestBase {
```

Comment:
Also added `TwoPhaseCommitTransactionIntegrationTestBase` and its implementation classes for the storage adapters.